### PR TITLE
products - search by phrase

### PR DESCRIPTION
**Pull request by bionanek**

### CODE REVIEW

File: api/controllers/productsController.js [hunk 1]
```diff
@@ -46,10 +46,20 @@ exports.getAll = async (request, response, next) => {
...
     }
 };
+
+exports.getSearchResults = async (request, response) => {
+    try {
+        const filteredProducts = await productsDomain.getSearchResults(request.query.search);
+
+        response.status(200).json(filteredProducts);
+    } catch (erorr) {
+
```
Comment (kachmashk) [inline api/controllers/productsController.js]: empty error

File: api/routes/validation/productValidation.js [hunk 2]
```diff
@@ -13,21 +13,37 @@ class ProductValidation {
...
                     break;
                 case "GET":
-                    ProductValidation.ValidateId(request.params.productId);
+                    ProductValidation.ValidateGet(request);
                     break;
                 case "PATCH":
                     ProductValidation.ValidateUpdate(request);
                     break;
                 case "DELETE":
                     ProductValidation.ValidateId(request.params.productId);
```
Comment (kachmashk) [inline api/routes/validation/productValidation.js]: should be ValidateIdIsNaN

File: api/tests/validation/productValidation.test.js [hunk 3]
```diff
@@ -124,6 +124,35 @@ describe('Products: ValidateIdExists', () => {
...
+            .to.has.property('message', 
+                'Search query is missing \'search\' field. Correct query should looke like this: \'/products/search?search=yourRequestedSearchPhrase\'');
+    });
+
+    it('Should pass validation with correct request query', () => {
+        const searchRequest = {
+                search: "search query"
+        }
+        
+
```
Comment (kachmashk) [inline api/tests/validation/productValidation.test.js:147]: remove unnecessary blank lines

File: api/controllers/productsController.js (hunk 1 above)
Comment (bionanek) [reply api/controllers/productsController.js]: dun

File: api/routes/validation/productValidation.js (hunk 2 above)
Comment (bionanek) [reply api/routes/validation/productValidation.js]: dun

File: api/tests/validation/productValidation.test.js (hunk 3 above)
Comment (bionanek) [reply api/tests/validation/productValidation.test.js:147]: dun

**Review by kachmashk — APPROVED** [review]

**Pull request merged**